kustomize/resource_kustomization.go

```diff
@@ -182,12 +182,19 @@ func kustomizationResourceDiff(ctx context.Context, d *schema.ResourceDiff, m in
 	mapper := m.(*Config).Mapper
 	gzipLastAppliedConfig := m.(*Config).GzipLastAppliedConfig
 
+	// Diff Old/Modified
 	do, dm := d.GetChange("manifest")
+	tm := dm.(string)
+	// dm will always be empty if it's known after apply
+	if tm == "" {
+		return nil
+	}
```

Comment:
I will have to look into this to fully understand it before I can merge.

Reply:
You can quite easily replicate the issue by creating something like:
1. `resource "random_id"`
2. `data "kustomization_overlay" "prerender"`
3. `data "kustomization_overlay" "source"` - make this depend on the `random_id`
4. `resource "kustomization_resource"` depending on ids from `prerender` and manifests from `source`